src/chess/position.h

```diff
@@ -77,7 +77,9 @@ class Position {
   int ply_count_ = 0;
 };
 
-enum class GameResult { UNDECIDED, WHITE_WON, DRAW, BLACK_WON };
+// These are ordered so max() prefers the best result.
+enum class GameResult : uint8_t { UNDECIDED, BLACK_WON, DRAW, WHITE_WON };
+GameResult operator-(const GameResult& res);
```

Comment:
That's a bit of an abuse of operator overload. :)

Reply:
As in revert this to the lambda I had before:
https://github.com/LeelaChessZero/lc0/commit/99abcc1587ce7734e75e9fa6941173cfbe7a7590#diff-83b7141f4fa67bcfdf95188467895cb5R1536-R1540
```c
  const auto FlipResult = [](const GameResult& res) {
    return res == GameResult::BLACK_WON ? …
```
Or somehow otherwise defined?

Reply:
Probably a function with name `FlipGameResult` is better (not as lambda, but here), but if you'd like to keep `operator-`, it's also fine.